src/fvmcts/fv_mcts_vanilla.jl

```diff
@@ -0,0 +1,367 @@
+## We can use the following directly without modification
+## 1. domain_knowledge.jl for Rollout, init_Q and init_N functions
+## 2. FVMCTSSolver for representing the overall MCTS (underlying things will change)
+using StaticArrays
+using Parameters
+using Base.Threads: @spawn
+
+abstract type AbstractCoordinationStrategy end
+
+struct VarEl <: AbstractCoordinationStrategy
+end
+
+@with_kw struct MaxPlus <:AbstractCoordinationStrategy
+    message_iters::Int64 = 10
+    message_norm::Bool = true
+    use_agent_utils::Bool = false
+    node_exploration::Bool = true
+    edge_exploration::Bool = true
+end
+
+@with_kw mutable struct FVMCTSSolver <: AbstractMCTSSolver
```

Comment:
Note to self: Add docu ala https://github.com/JuliaPOMDP/MCTS.jl/blob/a24e92d108853a0410ac0a61f64d4e53a8623e71/src/vanilla.jl#L1-L52